libs/content/workspace/src/tab/markdown_editor/images.rs

```diff
@@ -130,9 +68,19 @@ pub fn calc(
                             *image_state.lock().unwrap() = ImageState::Failed(err);
                         }
                     }
+                });
 
-                    // request a frame when the image is done loading
-                    ctx.request_repaint();
+                #[cfg(not(target_arch = "wasm32"))]
+                thread::spawn(move || {
```

Comment:
Could we have used something from `futures`, the way you managed the blocking behavior in lb-rs, to make the functions in this file async whether they're wasm or not? Then we wouldn't need two versions and I imagine this diff would be much cleaner